Kernel/Process.cpp

```diff
@@ -567,6 +567,11 @@ void kgettimeofday(timeval& tv)
     tv = kgettimeofday();
 }
 
+void ksettimeofday(timeval tv)
+{
+    g_timeofday = tv;
```

Comment:
This doesn't seem like it would work. `g_timeofday` will be overwritten on next timer IRQ anyway.
Also, is this meaningfully different from `sys$clock_settime()` (which we already have)?

Reply:
Oh, that's how this works. Why does kgettimeofday() not return TimeManagement::the().epoch_time()? Concurrency?

Changed things to make this a LibC-only implementation.

Reply:
Yeah basically. And the idea going forward is to move the timestamp to a single canonical location that is also accessible from userspace via a special mapping.